[ci] shard distributed on cuda 10.2

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #77276

We shareded the pull request distributed job, should do this one as well

suo committed:
[ci] shard distributed on cuda 10.2

We shareded the pull request distributed job, should do this one as well

[ghstack-poisoned]

janeyx99 commented on .github/workflows/trunk.yml:
```diff
@@ -67,6 +67,7 @@ jobs:
           { config: "nogpu_NO_AVX2", shard: 1, num_shards: 1, runner: "linux.2xlarge" },
           { config: "jit_legacy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
           { config: "distributed", shard: 1, num_shards: 1, runner: "linux.8xlarge.nvidia.gpu" },
```
```suggestion
          { config: "distributed", shard: 1, num_shards: 2, runner: "linux.8xlarge.nvidia.gpu" },
```

suo committed:
Update on "[ci] shard distributed on cuda 10.2"


We shareded the pull request distributed job, should do this one as well

[ghstack-poisoned]

suo commented:
@pytorchbot merge this on green

janeyx99 commented:
@pytorchbot merge this on green
